app.py

```diff
@@ -258,15 +262,32 @@ def get_chart():
                 scaleratio=1,
             )
         )
-  
+
+ 
         fig.update_layout(height=600, width=800)
         # Add header
         st.markdown("<h3 style='text-align: center; color: black;'>Marriage Incentive and Penalty Analysis</h3>", unsafe_allow_html=True)
         fig = format_fig(fig)
         # Display the chart
         
         st.plotly_chart(fig, use_container_width=True)
-    
-    get_chart()
+@st.cache_data()
```

Comment:
need to investigate why the heatmap output differs from the table output - I believe the table is mostly aligned with the App but the heatmap seems off in all of the cases